cmd/prom2json/main.go

```diff
@@ -26,37 +28,55 @@ import (
 	"github.com/prometheus/prom2json"
 )
 
+var USAGE = fmt.Sprintf(`Usage: %s [METRICS_PATH | METRICS_URL [--cert CERT_PATH --key KEY_PATH | --accept-invalid-cert]]`, os.Args[0])
+
 func main() {
-	cert := flag.String("cert", "", "certificate file")
-	key := flag.String("key", "", "key file")
+	cert := flag.String("cert", "", "client certificate file")
+	key := flag.String("key", "", "client certificate's key file")
 	skipServerCertCheck := flag.Bool("accept-invalid-cert", false, "Accept any certificate during TLS handshake. Insecure, use only for testing.")
 	flag.Parse()
 
-	if len(flag.Args()) != 1 {
-		log.Fatalf("Usage: %s METRICS_URL", os.Args[0])
-	}
-	if (*cert != "" && *key == "") || (*cert == "" && *key != "") {
-		log.Fatalf("Usage: %s METRICS_URL\n with TLS client authentication: %s -cert=/path/to/certificate -key=/path/to/key METRICS_URL", os.Args[0], os.Args[0])
+	var input io.Reader
+	var err error
+	arg := flag.Arg(0)
+
+	if arg == "" {
+		input = os.Stdin
+	} else if URL, urlErr := url.Parse(arg); urlErr != nil || URL.Scheme == "file" || URL.Scheme == "" {
+		// Try to parse the arg as an url
+		if input, err = os.Open(arg); err != nil {
```

Comment:
This would pass a `file:///foo/bar` URL directly to `os.Open`. That's not going to work, is it?
In doubt, I would just not handle the `file://` scheme, i.e. let such a URL be handled by `prom2json.FetchMetricFamilies`, which will then result in an error: unsupported protocol scheme "file"